node/npm_import_error.ts

```diff
@@ -0,0 +1,23 @@
+class NPMImportError extends Error {
+  constructor(originalError: Error, public readonly moduleName: string) {
+    super(`It seems like you're trying to import an NPM module, did you mean to 'import ... from "npm:${moduleName}"'?`)
+
+    this.name = 'NPMImportError'
+    this.stack = originalError.stack
+
+    // https://github.com/microsoft/TypeScript-wiki/blob/8a66ecaf77118de456f7cd9c56848a40fe29b9b4/Breaking-Changes.md#implicit-any-error-raised-for-un-annotated-callback-arguments-with-no-matching-overload-arguments
+    Object.setPrototypeOf(this, NPMImportError.prototype)
+  }
+}
+
+const checkNpmImportError = (input: unknown): NPMImportError | undefined => {
+  if (input instanceof Error) {
+    const match = input.message.match(/Relative import path "(.*)" not prefixed with/)
```

Comment:
Matching the error message feels a bit brittle. As I suggested [here](https://github.com/netlify/pod-compute/issues/287#issuecomment-1314520156), we could leverage the fact that we're in control of the module resolving (in production anyway, where we use ESZIP).

We should be able to look at a specifier and see whether it's a full URL, a relative path, or anything else. If it's _anything else_ and the loading fails, I think we can safely infer that people are either trying to load an npm module or a specifier that should be in the import map but isn't.

I think doing that check in the loader would be more robust.

Reply:
Ah, interesting! I thought the ESZIP loader would be called even for paths that don't match a URL format or a path. In that case, ignore me!